src/k8s/cmd/k8s/k8s_helm.go

```diff
@@ -15,6 +17,17 @@ var (
 		Hidden:             true,
 		DisableFlagParsing: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			snap := snap.NewSnap(os.Getenv("SNAP"), os.Getenv("SNAP_COMMON"))
+
+			isWorker, err := snaputil.IsWorker(snap)
+			if err != nil {
+				return fmt.Errorf("failed to check if node is a worker: %w", err)
+			}
+
+			if isWorker {
+				return fmt.Errorf("this action is restricted on workers")
```

Comment:
Can we have a sentinel error for this? e.g. `ErrDisallowed` or `ErrControlPlaneOnly` (naming is hard). (A TODO is also fine.)
Background: I'm currently working on a PR to resolve errors to human cli errors.

Reply:
I've put a TODO, I can address this after your PR is merged. Depending on which goes in first, feel free to ping me. Thanks!